### PR TITLE
docs: tell typecheckers like pyright that Config.__getattr__ returns Any type

### LINKED ISSUE #7978: Typing issue with set_http_meta
### Summary of problem

`set_http_meta` seems to be typed too strictly to accept anything in existing contributed code.

`_JSONType | Unknown` is incompatible with the `IntegrationConfig` type

For example the falcon integration.

https://github.com/DataDog/dd-trace-py/blob/4577f757cc8ab750d1fdbfae1fb767dcbacf6f22/ddtrace/contrib/falcon/middleware.py#L49C1-L51C10

`pyright/pylance` will error here with

```
Argument of type "_JSONType | Unknown" cannot be assigned to parameter "integration_config" of type "IntegrationConfig" in function "set_http_meta"
  Type "_JSONType | Unknown" cannot be assigned to type "IntegrationConfig"
    "float" is incompatible with "IntegrationConfig"
```
### Which version of dd-trace-py are you using?
2.4.0
### Which version of pip are you using?
23.3

### How can we reproduce your problem?

VSCode + pylance / `npx pyright` / grab pyright from pypi and run it

### PR DESCRIPTION
**Pull request by emmettbutler**
This change fixes #7978 by being explicit about the return type of `Config.__getattr__`, avoiding linting failures arising from ambiguity.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/settings/config.py [hunk 1]
```diff
@@ -517,7 +517,7 @@ def __init__(self):
         self._telemetry_install_type = os.getenv("DD_INSTRUMENTATION_INSTALL_TYPE", None)
         self._telemetry_install_time = os.getenv("DD_INSTRUMENTATION_INSTALL_TIME", None)
 
-    def __getattr__(self, name):
+    def __getattr__(self, name) -> Any:
```
Comment (P403n1x87) [inline ddtrace/settings/config.py:520]: 🤔  can't this type be made more specific?

Comment (emmettbutler) [reply ddtrace/settings/config.py:520]: Probably yes - it has to encapsulate all of the types stored as attributes on the Config object.

Comment (mabdinur) [reply ddtrace/settings/config.py:520]: We could probably restrict it to IntegrationConfig and _JSONType

**Review by mabdinur — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

File: ddtrace/settings/config.py (hunk 1 above)
Comment (emmettbutler) [reply ddtrace/settings/config.py:520]: I might address this in a future pull request

**Pull request merged**